app/src/main/java/com/kickstarter/services/KSApolloClientV2.kt

```diff
@@ -380,42 +388,38 @@ class KSApolloClientV2(val service: ApolloClient, val gson: Gson) : ApolloClient
         }
     }
 
-    override fun getStoredCards(): Observable<List<StoredCard>> {
-        return Observable.defer {
-            val ps = PublishSubject.create<List<StoredCard>>()
+    override suspend fun _getStoredCards(): List<StoredCard> {
+        val query = UserPaymentsQuery()
+        var ret = listOf<StoredCard>()
+        try {
+            // `single()` vs `singleOrNull()`
+            val response = this.service.query(query).toFlow().single()
 
-            val query = UserPaymentsQuery()
-            this.service
-                .query(query)
-                .rxSingle()
-                .doOnError { throwable ->
-                    ps.onError(throwable)
-                }
-                .subscribe { response ->
-                    if (response.hasErrors()) {
-                        ps.onError(Exception(response.errors?.first()?.message))
-                    } else {
-                        val cardsList = mutableListOf<StoredCard>()
-                        response.data?.me?.storedCards?.nodes?.map {
-                            it?.let { cardData ->
-                                val card = StoredCard.builder()
-                                    .expiration(cardData.expirationDate)
-                                    .id(cardData.id)
-                                    .lastFourDigits(cardData.lastFour)
-                                    .type(it.type)
-                                    .stripeCardId(it.stripeCardId)
-                                    .build()
-                                cardsList.add(card)
-                            }
-                        }
-                        ps.onNext(cardsList)
-                    }
-                    ps.onComplete()
-                }.addToDisposable(disposables)
-            return@defer ps
+            if (response.hasErrors()) {
+                // parse API errors into specific type
+                throw Exception(response.errors?.first()?.message)
+            }
+
+            ret = response.data?.me?.storedCards?.nodes?.filterNotNull()?.map { cardData ->
+                StoredCard.builder()
+                    .expiration(cardData.expirationDate)
+                    .id(cardData.id)
+                    .lastFourDigits(cardData.lastFour)
+                    .type(cardData.type)
+                    .stripeCardId(cardData.stripeCardId)
+                    .build()
+            } ?: listOf()
+        } catch (apolloException: ApolloException) {
+            // parse network errors into specific type
```

Comment:
If we wanted to provide some feedback to the user. For example a pretty used pattern now  within the app is displaying a red snackbar on the UI, usually with a generic message "something went wrong if no message provided on the API". It was achieved before with the `doOnError` block.
We could add some kind of callback the consumer of `_getStoredCards()` to still be able to provide feedback to the user when a error happens.